docs/examples/use_cases/pytorch/single_stage_detector/src/coco_pipeline.py

```diff
@@ -15,110 +15,84 @@
 
 import torch
 from nvidia.dali.pipeline import Pipeline
-import nvidia.dali.ops as ops
 import nvidia.dali.types as types
+import nvidia.dali.fn as fn
+
+
+def create_coco_pipeline(default_boxes, args, seed):
+    pipeline = Pipeline(args.batch_size, args.num_workers,
+                        args.local_rank, seed=seed)
+    try:
+        shard_id = torch.distributed.get_rank()
+        num_shards = torch.distributed.get_world_size()
+    except RuntimeError:
+        shard_id = 0
+        num_shards = 1
+    with pipeline:
+        images, bboxes, labels = fn.coco_reader(file_root=args.train_coco_root,
+                                                annotations_file=args.train_annotate,
+                                                skip_empty=True,
+                                                shard_id=shard_id,
+                                                num_shards=num_shards,
+                                                ratio=True,
+                                                ltrb=True,
+                                                random_shuffle=False,
+                                                shuffle_after_epoch=True,
+                                                name="Reader")
+
+        crop_begin, crop_size, bboxes, labels = fn.random_bbox_crop(bboxes, labels,
+                                                                    device="cpu",
+                                                                    aspect_ratio=[
+                                                                        0.5, 2.0],
+                                                                    thresholds=[
+                                                                        0, 0.1, 0.3, 0.5, 0.7, 0.9],
+                                                                    scaling=[
+                                                                        0.3, 1.0],
+                                                                    bbox_layout="xyXY",
+                                                                    allow_no_crop=True,
+                                                                    num_attempts=1)
```

Comment:
I wonder if the number of attempts being 1 was intentional in the first place.

Reply:
Copy-paste. Done